text/0045-api.md

```diff
@@ -0,0 +1,191 @@
++ Feature name: `api`
++ Start date: 2020-07-14
++ RFC PR: [iotaledger/bee-rfcs#45](https://github.com/iotaledger/bee-rfcs/pull/45)
++ Bee issue: [iotaledger/bee#105](https://github.com/iotaledger/bee/issues/105)
+
+# Summary
+
+This RFC proposes a crate that provides interfaces and tools to support presentation layers such as web, mobile, or UI. The focus of this RFC therefore lies on high-level interaction.
+
+# Motivation
+
+To be able to program applications that interface with the Bee node, an Application Programming Interface (API) is needed. An application might want to fetch information about the status of the Bee node, look up certain transactions in  the tangle, or be able to submit transactions on its own.
+With the help of this interface, applications can interact with the Bee node and participate in the IOTA network.
+
+# Detailed design
+
+The design describes a layered approach that provides services on which applications can build.
+
+## Service
+
+The `Service` trait defines all the services that a Bee node exposes. It hides internal implementation and changes, furthermore governs external access to data and functions. It can support multiple presentation layers such as web, mobile, UI, etc. 
+Since it hides internal logic and governs access, it helps maintain the layers above.
+The `Service` trait is defined as follows:
```

Comment:
Thats a really good question. Will add this to the open questions.
API is a big term, it could mean low-level api where libraries get loaded or higher-level ones such as Web, UI, etc.
Whats sure, we should not try to mix all that up with one API trait, it will probably quite hard to maintain and isn't clean.

MQTT is currently developed in it's own crate. Maybe we should put it then into bee-api.
or we should have independend crates and rename `bee-api` in something like `bee-web-service`